antarest/storage/business/study_download_utils.py

```diff
@@ -0,0 +1,319 @@
+            files_matcher = (
+                [f"values-{str(data.level)}", f"details-{str(data.level)}"]
+                if data.includeClusters
+                else [f"values-{str(data.level)}"]
+            )
+            for elm in files_matcher:
+                tmp_url = f"{url}/{elm}"
+                StudyDownloader.read_columns(
+                    matrix, prefix, year, area_name, study, tmp_url, data
+                )
+
+    @staticmethod
+    def apply_type_filters(
+        matrix: MatrixAggregationResult,
+        prefix: str,
+        year: int,
+        type_elm: Any,
+        elm: str,
+        study: Study,
+        url: str,
+        data: StudyDownloadDTO,
+        first_element_type_condition: bool,
+        second_element_type_condition: Callable[[str], bool],
+    ) -> None:
+
+        if first_element_type_condition:
+            if StudyDownloadType.from_dict(
+                data.type
+            ) == StudyDownloadType.LINK and isinstance(type_elm[elm], Area):
+                if type_elm[elm].links:
+                    for out_link in type_elm[elm].links.keys():
+                        if second_element_type_condition(out_link):
+                            tmp_prefix = f"{prefix}|{elm}^{out_link}"
+                            link_url = f"{url}/{out_link}"
+                            StudyDownloader.level_output_filter(
+                                matrix,
+                                tmp_prefix,
+                                year,
+                                f"{elm}^{out_link}",
+                                study,
+                                link_url,
+                                data,
+                            )
+            else:
+                tmp_prefix = f"{prefix}|{elm}"
+                StudyDownloader.level_output_filter(
+                    matrix, tmp_prefix, year, elm, study, url, data
+                )
+
+    @staticmethod
+    def get_filters(filter: str) -> Tuple[str, str]:
+        if filter and len(filter) > 0:
+            tmp_filters = filter.split(">")
+            if len(tmp_filters) >= 2:
+                return tmp_filters[0], tmp_filters[1]
+            else:
+                return filter, ""
+        return "", ""
+
+    @staticmethod
+    def select_filter(
+        matrix: MatrixAggregationResult,
+        prefix: str,
+        year: int,
+        type_elm: Any,
+        study: Study,
+        url: str,
+        data: StudyDownloadDTO,
+    ) -> None:
+        for elm in type_elm.keys():
+            filtered_url = (
+                f"{url}/@ {elm}"
+                if StudyDownloadType.from_dict(data.type)
+                == StudyDownloadType.CLUSTER
+                else f"{url}/{elm}"
+            )
+
+            # Filter has priority over FilterIn or FilterOut
+            if data.filter and len(data.filter) > 0:
+                for flt in data.filter:
+                    tmp_filters = flt.split(">")
```

Comment:
add a comment explaining that only links can have a ">" character (referring to the "transform_name_to_id" method)